app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ResetAppUseCase.kt

```diff
@@ -0,0 +1,40 @@
+package co.netguru.baby.monitor.client.feature.settings
+
+import co.netguru.baby.monitor.client.common.NotificationHandler
+import co.netguru.baby.monitor.client.data.DataRepository
+import co.netguru.baby.monitor.client.data.splash.AppState
+import co.netguru.baby.monitor.client.feature.firebasenotification.FirebaseInstanceManager
+import io.reactivex.Completable
+import io.reactivex.rxkotlin.toCompletable
+import javax.inject.Inject
+
+class ResetAppUseCase @Inject constructor(
+    private val notificationHandler: NotificationHandler,
+    private val firebaseInstanceManager: FirebaseInstanceManager,
+    private val dataRepository: DataRepository
+) {
+
+    fun resetApp(): Completable {
+        return Completable.merge(
+            listOf(
+                handleAppState(),
+                dataRepository.deleteAllData(),
+                notificationHandler::clearNotifications.toCompletable()
+            )
+        )
+    }
+
+    private fun handleAppState(): Completable {
+        return dataRepository.getSavedState()
+            .flatMapCompletable { appState ->
+                when (appState) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/59ecd15255bdd529a391ea0d5072dc37d88e9138/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/settings/ResetAppUseCase.kt#L30"></span>Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>